src/dialog/dialog.scss

```diff
@@ -0,0 +1,207 @@
+// Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+// See LICENSE.md in the project root for license terms and full copyright notice.
+@import '../style/index';
+@import '../icon/index';
+
+@mixin iui-dialog-backdrop {
+  @include iui-reset;
+  z-index: 999;
+  isolation: isolate;
+  position: fixed;
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  @include themed {
+    background-color: rgba(0, 0, 0, t(iui-opacity-4));
+  }
+
+  // Small/Resizable Dialog Animations for CSS
+  visibility: hidden;
+  opacity: 0;
+  transition: visibility $iui-speed-instant linear $iui-speed-fast, opacity $iui-speed-fast ease-out;
+
+  &.iui-dialog-visible {
+    visibility: visible;
+    opacity: 1;
+    // remove delay for entry animation so dialog is instantly visible
+    transition-delay: $iui-speed-instant;
+  }
+}
+
+@mixin iui-dialog {
+  border-radius: $iui-border-radius;
+  box-shadow: $iui-elevation-24;
+  box-sizing: border-box;
+  position: absolute;
+  padding: $iui-baseline $iui-m;
+  box-sizing: border-box;
+  padding: $iui-baseline $iui-m;
+  overflow: hidden;
+  @include themed {
+    background-color: t(iui-color-background-1);
+  }
+}
+
+@mixin iui-dialog-default {
+  .iui-dialog {
+    left: 50%;
+    top: 33%;
+    transform: translate(-50%, -33%);
+    max-width: 50%;
+    min-width: 380px;
+    max-height: 100vh;
+
+    @media screen and (max-width: 400px) {
+      max-width: 95%;
+      width: 95%;
+      min-width: 95%;
+    }
+  }
+}
+
+@mixin iui-dialog-full-page {
+  @media (prefers-reduced-motion: no-preference) {
+    transition: visibility $iui-speed-instant linear $iui-speed-slow, opacity $iui-speed ease-out $iui-speed-fast;
+  }
+
+  .iui-dialog {
+    border-radius: 0;
+    height: 100vh;
+    width: 100vw;
+    top: 0;
+    left: 0;
+    max-width: initial;
+    min-width: initial;
+    display: flex;
+    flex-direction: column;
+    will-change: transform;
+    @media (prefers-reduced-motion: no-preference) {
+      transform: translateY(100%);
+      transition: visibility $iui-speed-instant linear $iui-speed, opacity $iui-speed-instant linear $iui-speed,
+        transform $iui-speed-fast ease-in;
+    }
+  }
+
+  &.iui-dialog-visible .iui-dialog {
+    transform: translateY(0);
+    @media (prefers-reduced-motion: no-preference) {
+      transition: transform $iui-speed ease-out;
+    }
+  }
+}
+
+@mixin iui-dialog-draggable {
```

Comment:
We might want to change to `position: absolute` to avoid dialogs going out of containers e.g. having app with a viewer:
![image](https://user-images.githubusercontent.com/36186912/167080849-f018fdf9-9761-4509-ad6a-9b3d6f073350.png)
![image](https://user-images.githubusercontent.com/36186912/167080823-016a3435-867d-4d9a-98db-8e68302c5950.png)
Because otherwise dialog would go on sidenav and header.

Reply:
there is position absolute on `iui-dialog`

Reply:
But it isn't helping, I guess because parent backdrop is placed on the whole screen with `position: fixed`